internal/declarative/v2/reconciler.go

```diff
@@ -377,17 +381,18 @@ func (r *Reconciler) pruneDiff(
 	return renderer.RemovePrerequisites(ctx, obj)
 }
 
-func pruneKymaSystem(diff []*resource.Info) []*resource.Info {
+func pruneResource(diff []*resource.Info, resourceType string, resourceName string) []*resource.Info {
 	for i, info := range diff { //nolint:varnamelen
 		obj := info.Object.(client.Object)
-		if obj.GetObjectKind().GroupVersionKind().Kind != "Namespace" {
+		if obj.GetObjectKind().GroupVersionKind().Kind != resourceType {
 			continue
 		}
-		if obj.GetName() != namespaceNotBeRemoved {
+		if obj.GetName() != resourceName {
 			continue
 		}
 		return append(diff[:i], diff[i+1:]...)
```

Comment:
This line is executed only if: `obj.GetObjectKind().GroupVersionKind().Kind == resourceType && obj.GetName() == resourceName`
Why not replace the lines above that use negated conditions and loop skipping with such an explicit check?